BITSWAP.md

```diff
@@ -11,56 +12,51 @@
+Bitswap is a data exchange protocol for sending and receiving content addressed blocks of data. Bitswap has two primary jobs:
 1. Attempt to acquire blocks from the network that have been requested by the client.
-2. Judiciously (though strategically) send blocks in its possession to other peers who want them.
+2. Send blocks in its possession to other peers who want them.
 
 # Organization of this document
 
 - [Introduction](#introduction)
-- [Subsystems](#subsystems)
-- [Implementation Details](#implementation-details)
-- [API Spec](#api-spec)
+- [Bitswap Protocol Versions](#bitswap-protocol-versions)
+   - [Bitswap 1.0.0](#bitswap-100)
+   - [Bitswap 1.1.0](#bitswap-110)
+   - [Bitswap 1.2.0](#bitswap-120)
 - [Implementations](#implementations)
 
 # Introduction
 
-Bitswap is IPFS’s central block exchange protocol. It handles the requests made by an IPFS user, human, or application to fetch data blocks from the network. It interacts with other Bitswap agents present in other IPFS nodes, exchanging (fetching + serving) blocks as it needs.
+Bitswap is a message based protocol, as opposed to response-reply. All messages contain wantlists, and/or blocks. Upon receiving a wantlist, a Bitswap server should consider sending out wanted blocks if it has them. Upon receiving blocks, the node should send out a notification called a `Cancel` to any other peers they have already asked for the data signifying that they no longer want the block. At the protocol level, Bitswap is very simple.
 
-Bitswap is a message based protocol, as opposed to response-reply. All messages contain wantlists, or blocks. Upon receiving a wantlist, an IPFS node should consider sending out wanted blocks if it has them. Upon receiving blocks, the node should send out a notification called a `Cancel` signifying that they no longer want the block. At the protocol level, Bitswap is very simple.
+Bitswap is a relatively simple protocol, where complexity lives in the particular implementations to optimize for their particular balance of aspects such as throughput, latency, fairness, memory usage, etc.
 
-While Bitswap is a relatively simple protocol, a time- and memory- performant implementation requires that many details be carefully thought out. We aim to document these details here so that future implementers may build upon the knowledge gathered over the several iterations of Bitswap.
+# Bitswap Protocol Versions
 
-# Subsystems
+There are multiple Bitswap versions and more may evolve over time. We give brief overviews as to the changes behind each protocol version.
 
-There are two primary flows that Bitswap manages: requesting blocks from and serving blocks to peers. Block requests are primarily mediated by the want-manager, which tells our peers whenever we want a new block. Serving blocks is primarily handled by the decision engine, which decides how resources should be allocated among our peers.
+- `/ipfs/bitswap/1.0.0` - Initial version
```

Comment:
I've also seen `/ipfs/bitswap` and `/bitswap/1.0.0` around digging through the GitHub history. I'm not sure if these are the same as `/ipfs/bitswap/1.0.0` or different earlier versions.

Perhaps someone with sufficient history can clarify, or I can do more GitHub archaeology to learn more.